tests/test_repositories.py

```diff
@@ -88,7 +159,33 @@ def test_create_remote_repository_fail_if_user_already_exists(mocker):
     with pytest.raises(RepositoryAlreadyExistsException):
         artifactory_repo.create_remote_repo(REMOTE_REPOSITORY)
 
-        artifactory_repo.get_remote_repo.assert_called_once_with(REMOTE_REPOSITORY.key)
+    artifactory_repo.get_remote_repo.assert_called_once_with(REMOTE_REPOSITORY.key)
+
+
+@responses.activate
+def test_create_local_repository_using_create_repo_success(mocker):
+    responses.add(
+        responses.GET, f"{URL}/api/repositories/{LOCAL_REPOSITORY.key}", status=404
+    )
+    responses.add(
+        responses.PUT,
+        f"{URL}/api/repositories/{LOCAL_REPOSITORY.key}",
+        json=LOCAL_REPOSITORY_RESPONSE.dict(),
+        status=201,
+    )
+    responses.add(
+        responses.GET,
+        f"{URL}/api/repositories/{LOCAL_REPOSITORY.key}",
+        json=LOCAL_REPOSITORY_RESPONSE.dict(),
+        status=200,
+    )
+
+    artifactory_repo = ArtifactoryRepository(AuthModel(url=URL, auth=AUTH))
+    mocker.spy(artifactory_repo, "get_repo")
+    local_repo = artifactory_repo.create_repo(LOCAL_REPOSITORY)
+    artifactory_repo.get_repo.side_effect = RepositoryNotFoundException

```

Comment:
What are you doing here? You're changing the behavior of a method you spy on? Or you forgot to assert?

Reply:
The create_repo is successful only if the get_repo returns a RepositoryNotFoundException. That's why I do that

Reply:
At this stage the get_repo has already been called (in the create_repo just above), so changing it now doesn't do anything. Furthermore, you cannot change the behavior of a *spied* method, the whole point is to let it do what it's supposed to do, if you want to change the behavior you mock it. Finally the get_repo method already raises the exception thanks to the first `responses` you added.